tests/platform_tests/fwutil/fwutil_common.py

```diff
@@ -0,0 +1,228 @@
+
+def complete_install(duthost, localhost, boot_type, res, pdu_ctrl, auto_reboot=False, current=None, next_image=None):
+    hn = duthost.mgmt_ip
+
+    if boot_type != "none":
+        if not auto_reboot:
+            logger.info("Waiting on install to finish.")
+            res.get(TIMEOUT)
+            logger.info("Rebooting switch using {} boot".format(boot_type))
+            duthost.command("sonic-installer set-default {}".format(current))
+            reboot(duthost, pdu_ctrl, boot_type)
+        
+        logger.info("Waiting on switch to shutdown...")
+        # Wait for ssh flap
+        localhost.wait_for(host=hn, port=22, state='stopped', delay=10, timeout=TIMEOUT)
+        logger.info("Letting switch get through ONIE / BIOS before pinging....")
+        time.sleep(300)
+        logger.info("Waiting on switch to come up....")
+        localhost.wait_for(host=hn, port=22, state='started', delay=10, timeout=300)
+        logger.info("Waiting on critical systems to come online...")
+        wait_until(300, 30, duthost.critical_services_fully_started)
+        time.sleep(60)
+
+    # Reboot back into original image if neccesary
+    if next_image:
+        logger.info("We booted into the new image... booting back into the image under test.")
+        duthost.command("sonic-installer set-default {}".format(current))
+        reboot(duthost, pdu_ctrl, "cold")
+        logger.info("Waiting on switch to shutdown....")
+        localhost.wait_for(host=hn, port=22, state='stopped', delay=10, timeout=150)
+        time.sleep(100)
+        logger.info("Waiting on switch to come up....")
+        localhost.wait_for(host=hn, port=22, state='started', delay=10, timeout=150)
+        wait_until(300, 30, duthost.critical_services_fully_started)
+        time.sleep(60)
+
+def show_firmware(duthost):
+    out = duthost.command("fwutil show status")
+    
+    num_spaces = 2
+    curr_chassis = ""
+    output_data = {"chassis":{}}
+    status_output = out['stdout']
+    separators = re.split(r'\s{2,}', status_output.splitlines()[1])  # get separators
+    output_lines = status_output.splitlines()[2:]
+
+    for line in output_lines:
+        data = []
+        start = 0
+
+        for sep in separators:
+            curr_len = len(sep)
+            data.append(line[start:start+curr_len].strip())
+            start += curr_len + num_spaces
+
+        if data[0].strip() != "":
+            curr_chassis = data[0].strip()
+            output_data["chassis"][curr_chassis] = {"component": {}}
+
+        output_data["chassis"][curr_chassis]["component"][data[2]] = data[3]
+
+    return output_data
+
+def get_install_paths(duthost, fw, versions, chassis):
+    chass = show_firmware(duthost)["chassis"].keys()[0]
+    component = fw["chassis"].get(chass, {})["component"]
+    ver = versions["chassis"].get(chass, {})["component"]
+    
+    paths = {}
+
+    for comp, revs in component.items():
+        if comp in ver:
```

Comment:
@alexrallen do we need a warning/error here in case such a situation takes place?

Reply:
Yes.